contrib/internal/redigo/redigo.go

```diff
@@ -0,0 +1,92 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package redigo
+
+import (
+	"net/url"
+	"reflect"
+	"regexp"
+	"strconv"
+	"unsafe"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+
+	garyburdredis "github.com/garyburd/redigo/redis"
+	gomoduleredis "github.com/gomodule/redigo/redis"
+)
+
+// RedisDialConfig contains the configurable parameters for the Redis connection.
+type RedisDialConfig struct {
+	DB int
+}
+
+const (
+	structFieldNameDialOptionFunc = "f"
+	structFieldNameConfigDBIndex  = "db"
```

Comment:
I feel very apprehensive about making changes like this... It's very brittle. In order for this to be safe in the future, we're essentially requiring that the upstreams don't change the names of their **unexported fields** within an exported struct, or otherwise refactoring their code in a way that this would break. There's no way we can guarantee that a package won't change their internal code (nor should we try to), and I don't think it's worth maintaining. Is there any precedence for this in any of our other contrib libraries?